frontend/ui/data/keyboardlayouts/fa_keyboard.lua

```diff
@@ -0,0 +1,126 @@
+local waw = fa_popup.waw
+local zay = fa_popup.zay
+local jee = fa_popup.jee
+local ha = fa_popup.ha     -- while this is Persian letter ح / as in the sound you make when blowing on a glass to clean it.
+local tah = fa_popup.tah
+local yaa = fa_popup.yaa
+local kaf = fa_popup.kaf
+local gaf = fa_popup.gaf
+local lam = fa_popup.lam
+local meem = fa_popup.meem
+local nun = fa_popup.nun
+local seen = fa_popup.seen
+local ayin = fa_popup.ayin
+local fah = fa_popup.fah
+local saad = fa_popup.saad
+local qaf = fa_popup.qaf
+local raa = fa_popup.raa
+local sheen = fa_popup.sheen
+local taa = fa_popup.taa
+local thaa = fa_popup.thaa
+local th_aa = fa_popup.th_aa
+local thaal = fa_popup.thaal
+local dhad = fa_popup.dhad
+local ghayn = fa_popup.ghayn
+local khaa = fa_popup.khaa
+local diacritics = fa_popup.diacritics
+local diacritic_fat_ha = fa_popup.diacritic_fat_ha
+local diacritic_damma = fa_popup.diacritic_damma
+local diacritic_kasra = fa_popup.diacritic_kasra
+local diacritic_sukoon = fa_popup.diacritic_sukoon
+local diacritic_shadda = fa_popup.diacritic_shadda
+local diacritic_tanween_fath = fa_popup.diacritic_tanween_fath
+local diacritic_tanween_damm = fa_popup.diacritic_tanween_damm
+local diacritic_tanween_kasr = fa_popup.diacritic_tanween_kasr
+local arabic_comma = fa_popup.arabic_comma
+
+
+return {
+    min_layer = 1,
+    max_layer = 4,
+    shiftmode_keys = {["1/2"] = true, ["2/2"] = true},                   
+    symbolmode_keys = {["نشانه‌ها"] = true,["الفبا"]=true},      -- نشانه‌ها means "Symbol", الفبا means "letter" (traditionally "ABC" on QWERTY layouts)
+    utf8mode_keys = {["🌐"] = true},                      -- The famous globe key for layout switching
+    umlautmode_keys = {["Äéß"] = false},                  -- No need for this keyboard panel
+    keys = {
+        -- first row
+        {  --  1                         2            3      4
+            { dhad,      				dhad,        "~",   "1", },
+            { saad,   					saad,        "`",   "2", },
+            { thaa,      				thaa,        "|",   "3", },
+            { qaf,  					qaf,         "•",   "4", },
+            { fah,                      fah,         "√",   "5", },
+            { ghayn,                    ghayn,       "π",   "6", },
+            { ayin,                     ayin,        "÷",   "7", },
+            { h_aa,                     h_aa,        "×",   "8", },
+            { khaa,                     khaa,        "¶",   "9",  },
+            { ha,                       ha,          "Δ",  "0",  },
+            { jeem,                     jeem,        "‘",   ">"  },
+        },
+        -- second row
+        {  --  1                         2            3       4
+            { sheen,        			sheen,       "£",    _at, },
+            { seen,    					seen,        "¥",    "#", },
+            { yaa,                      yaa,         "$",    "﷼", },
+            { ba,                       ba,          "¢",    "ـ", },
```

Comment:
Is there a particular reason you opted to have a fair number of characters (including ba) as a popup even though there's only one thing in it?

Reply:
in Persian we don't have upper case so i don't need Shift key in alphabet mode
in other hand if you press [1/2] it change to [2/2] and than press alphabet  it go to the Alphabet mode.
it is a hack to press "الفبا" change the keyboard to alphabet mode and ignore shift mode.
do you have a better solution for this problem?

Reply:
I think you misunderstood me slightly. The duplication doesn't bother me, but you could also write it like this:
```suggestion
            {  "ب",                        "ب",          "¢",    "ـ", },
```

Or keep this line the same, but write:

```
local ba = "ب"
```

I'm talking about the difference between a real popup:
![Screenshot_2021-01-31_21-23-48](https://user-images.githubusercontent.com/202757/106396808-ab38a980-640a-11eb-9a29-b03750a1f2fc.png)

And one that's technically a popup, but has only one entry:
![Screenshot_2021-01-31_21-24-01](https://user-images.githubusercontent.com/202757/106396814-b390e480-640a-11eb-9085-30d68f0f95f0.png)

Reply:
ok
i get your point now
i start work on this by review Arabic keyboard
https://github.com/koreader/koreader/blob/6c0345de17af187b8c87293ba84c8d7913891b83/frontend/ui/data/keyboardlayouts/keypopup/ar_AA_popup.lua#L18-L20
but you right, it's my fault. i will be fix that as soon as possible

Reply:
hi
i fix this issue with b21de3f3a60a300b42da470860171d77e41f522d and fee1d606d3fc964b19c74da1dc621fc87278a5e4 commit. if it is acceptable please resolve this conversation